src/toast/data.py

```diff
@@ -141,86 +166,167 @@ def _get_optional(k, dt):
                 wcomm.barrier()
         return
 
-    def select_observations(self, key, require=False):
-        """Given an observation key, return lists of observations with unique values.
+    def split(
```

Comment:
Any chance that the `split()` and `select()` methods could return data objects with broken communicators?

Reply:
As discussed offline, the select method now also has a check for consistent dictionary values across the group.